### PR TITLE
add RVC instr c.lw, c.ld, c.fld

### PR DESCRIPTION
**Pull request by derekztu22**

### CODE REVIEW

File: src/codegen/riscv64/assembler-riscv64.h [hunk 1]
```diff
@@ -602,6 +602,9 @@ class V8_EXPORT_PRIVATE Assembler : public AssemblerBase {
   void c_swsp(Register rs2, uint16_t uimm8);
   void c_sdsp(Register rs2, uint16_t uimm9);
   void c_fsdsp(FPURegister rs2, uint16_t uimm9);
+  void c_lw(Register rd, Register rs1, int16_t imm7);
+  void c_ld(Register rd, Register rs1, int16_t imm8);
+  void c_fld(FPURegister rd, Register rs1, int16_t imm8);
```
Comment (LOGQTAINIA) [inline src/codegen/riscv64/assembler-riscv64.h:607]: `int16_t` -> `uint16_t`

`imm7` -> `uimm7`

File: src/codegen/riscv64/assembler-riscv64.h [hunk 2]
```diff
@@ -1045,6 +1048,10 @@ class V8_EXPORT_PRIVATE Assembler : public AssemblerBase {
   void GenInstrCIU(uint8_t funct3, Opcode opcode, FPURegister rd, uint8_t uimm6);
   void GenInstrCSS(uint8_t funct3, Opcode opcode, FPURegister rs2, uint8_t uimm6);
   void GenInstrCSS(uint8_t funct3, Opcode opcode, Register rs2, uint8_t uimm6);
+  void GenInstrCL(uint8_t funct3, Opcode opcode, Register rd,
+                  Register rs1, int8_t uimm5);
+  void GenInstrCL(uint8_t funct3, Opcode opcode, FPURegister rd,
+                  Register rs1, int8_t uimm5);
```
Comment (LOGQTAINIA) [inline src/codegen/riscv64/assembler-riscv64.h:1054]: `int8_t` -> `uint8_t`

**Review by LOGQTAINIA — CHANGES_REQUESTED** [review]
Others look good to me.

File: src/codegen/riscv64/assembler-riscv64.cc [hunk 3]
```diff
@@ -905,6 +905,26 @@ void Assembler::GenInstrCSS(uint8_t funct3, Opcode opcode, FPURegister rs2,
...
+void Assembler::GenInstrCL(uint8_t funct3, Opcode opcode, FPURegister rd,
+                           Register rs1, int8_t uimm5) {
+  DCHECK(is_uint3(funct3) && rd.is_valid() && rs1.is_valid() && is_uint5(uimm5));
+  ShortInstr instr = opcode | ((uimm5 & 0x3) << 5) |
+	           ((rd.code() & 0x7) << kRvcRs2sShift) |
+                   ((uimm5 & 0x1c) << 8) | (funct3 << kRvcFunct3Shift) |
+                   ((rs1.code() & 0x7) << kRvcRs1sShift);
+  emit(instr);
+}
+
```
Comment (LOGQTAINIA) [inline src/codegen/riscv64/assembler-riscv64.cc:927]: the same as declaration.

`int8_t` -> `uint8_t`

File: src/codegen/riscv64/assembler-riscv64.cc [hunk 4]
```diff
@@ -1967,6 +1987,29 @@ void Assembler::c_fsdsp(FPURegister rs2, uint16_t uimm9) {
...
+  GenInstrCL(0b011, C0, rd, rs1, uimm5);
+}
+
+void Assembler::c_fld(FPURegister rd, Register rs1, int16_t uimm8) {
+  DCHECK(((rd.code() & 0b11000) == 0b01000) && ((rs1.code() & 0b11000) == 0b01000) &&
+        is_uint8(uimm8));
+  uint8_t uimm5 = ((uimm8 & 0x38) >>  1) | ((uimm8 & 0xc0) >> 6);
+  GenInstrCL(0b001, C0, rd, rs1, uimm5);
+}
+
```
Comment (LOGQTAINIA) [inline src/codegen/riscv64/assembler-riscv64.cc:2012]: the same as declaration.

`int16_t` -> `uint16_t`

**Review by LOGQTAINIA — CHANGES_REQUESTED** [review]
Keep consistance between declaration and definition

**Comment by derekztu22** [comment]
@penguinwu please let me know if you see any issues.

**Review by penguinwu — APPROVED** [review]
LGTM

**Pull request merged**